### ISSUE TITLE
Set label of the actual contained widget for main area widget (or no label at all)

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#17560]
### Problem

Since https://github.com/jupyterlab/jupyterlab/pull/17527 JupyterLab sets the `aria-label` on the `MainAreaWidget` as `main area content` and on its toolbar as `main area toolbar`:

https://github.com/jupyterlab/jupyterlab/blob/8401ef9900476b4d9445e19a41fd0a3ebadda125/packages/apputils/src/mainareawidget.ts#L47-L51

The "main area" is not a user-facing term in normal use and could lead to more confusion or annoyance than having no label at all.

### Proposed Solution

1. Investigate if this label is really needed or if the containing widgets always come with their own label
2. If it is needed, use the tile of the `MainAreaWidget` child to set a useful `aria-label` on the `MainAreaWidget` and its toolbar

### Additional context

- This was previously discussed in https://github.com/jupyterlab/jupyterlab/issues/13045
- JupyterLab 4.4.2

**Comment by andrii-i** [comment]
@krassowski thank you for opening this. Other than label not being user friendly/descriptive, is there also a problem of main area label and containing widgets label not working well both are present?

**Comment by krassowski** [comment]
> is there also a problem of main area label and containing widgets label not working well both are present?

I do not understand the question.

**Comment by andrii-i** [comment]
@krassowski edited the original question to add a missing word.

I was wondering what currently happens if both `aria-label` on the `MainAreaWidget` and `aria-label` on the widget contained within it are set.

**Comment by krassowski** [comment]
> I was wondering what currently happens if both aria-label on the `MainAreaWidget` and aria-label on the widget contained within it are set.

Good question. It looks like I was wrong, the `main area content` is not set on `MainAreaWidget` but on the widget contained within, overwriting any `aria-label` that the widget contained within may have had set. It feels like we should only set the `main area content` label if the node does not have `title` or `aria-label` already.

![Image](https://github.com/user-attachments/assets/53c9542f-98e6-41ec-834b-938e8f823c18)

Then the `MainAreaWidget` has `aria-labelled-by` which means it already gets the label from the title of the tab.
